crates/hir-expand/src/builtin_fn_macro.rs

```diff
@@ -247,160 +243,22 @@ fn format_args_expand_general(
     _db: &dyn ExpandDatabase,
     _id: MacroCallId,
     tt: &tt::Subtree,
-    end_string: &str,
+    // FIXME: Make use of this so that mir interpretation works properly
```

Comment:
Note, (I'm probably not fixing this in this PR) `format_args_nl!` will behave the same as `format_args!` that is no newline will be appended. This is fine for most hings but mir eval.

Reply:
I'm unsure how to solve this, we could give the construct an extra "parameter" or something that tells it to emit an additional newline, make a `builtin#format_args_nl` construct or try to fiddle with the template string which I really do not want to do (and I don't think we can reasonably do that).

Reply:
I'm leaning towards adding a second builtin# variant, it seems simplest